apps/backend/src/modules/users/user.service.ts

```diff
@@ -45,6 +52,13 @@ class UserService implements Service {
 		};
 	}
 
+	public async findByEmail(
+		email: UserEntity["email"],
+	): Promise<null | UserGetOneResponseDto> {
```

Comment:
```suggestion
	): Promise<null | UserEntity> {
```
Would that work?

Reply:
`Promise<null | UserEntity>` there gives an error:
```
Type '{ email: string; id: number; name: string; } | null' is not assignable to type 'UserEntity | null'.
  Type '{ email: string; id: number; name: string; }' is missing the following properties from type 'UserEntity': passwordHash, passwordSalt, toNewObject, toObject ts(2322)
```

Reply:
Keep it then. Also make sure we don't send passwordHash and passwordSalt to the client

Reply:
I've added `createdAt` and `updatedAt`, and tried to test returning  `Promise<null | UserEntity>` once more. However, the issue remains the same when I'm trying to return `Promise<null | UserEntity>`:

```
Type '{ createdAt: string; email: string; id: number; name: string; updatedAt: string; } | null' is not assignable to type 'UserEntity | null'.
  Type '{ createdAt: string; email: string; id: number; name: string; updatedAt: string; }' is missing the following properties from type 'UserEntity': passwordHash, passwordSalt, toNewObject, toObject  ts(2322)
```
I believe the difficulty is that I return `item.toObject()` in UserService.findByEmail (to avoid returning password hash and salt) and not an instance of UserEntity itself (`item` in my code ). It says that it "is missing the following properties from type 'UserEntity': passwordHash, passwordSalt, toNewObject, toObject". It doesn't miss createdAt and updatedAt.